commit f6518de961b3aaf0ac9f5ce3cb4a8f63e2dd76a9 (refs/changes/82/782/5)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2016-10-28T13:18:57+01:00 (2 years, 11 months ago)
    
    MANTA-2799 "manta-adm update" should fail when given non-existent service
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/lib/adm.js b/lib/adm.js
index 94e2b1a..39aa96c 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 /*
@@ -1390,6 +1390,11 @@ maAdm.prototype.generatePlan = function (callback, service, noreprovision)
 	var self = this;
 	var log = this.ma_log;
 
+	if (service && !svcs.serviceNameIsValid(service)) {
+		callback(new VError('unrecognized service: "%s"', service));
+		return;
+	}
+
 	log.info('generating plan');
 	log.trace('generating plan', this.ma_instances_wanted,
 	    this.ma_config_bycfg);
